lib/Log/Declare.pm

```diff
@@ -268,8 +284,12 @@ sub log {
                       ($ENV{'LOG_DECLARE_USE_LOCALTIME'} ? localtime : gmtime);
 
     $message .= "\n" if substr($message,-1) ne "\n";
+    my $log_message = "$$ [$ts] [$level_name]$categories $message";
+
+    return CORE::print STDERR $log_message if !defined $APPENDER;
 
-    return CORE::print STDERR "$$ [$ts] [$level_name]$categories $message";
+    my $log_method = lc($level_name);
+    return $APPENDER->$log_method($log_message);
```

Comment:
This can't work for structured logging. It's still supplying $log_message, which is a string.